### ISSUE TITLE
A comment at the beginning of SQL breaks the result of ActiveRecord::ConnectionAdapters::PostgreSQL::DatabaseStatements#write_query?

### ISSUE THREAD

**Issue by south37** [rails/rails#38933]
### Steps to reproduce

When using [marginalia gem](https://github.com/basecamp/marginalia) with `Marginalia::Comment.prepend_comment = true` option, comments are inserted at the beginning of SQL.
cf. https://github.com/basecamp/marginalia#prepend-comments

When using this option with `replica: true` setting, `ActiveRecord::ReadOnlyError` occurs when a reading query is thrown. The example is shown below.

```yaml
# config/database.yml
default:
.
.
.
  primary_replica:
    adapter: postgresql
    encoding: unicode
    url: <%= ENV['REPLICA_DATABASE_URL'].presence %>
    pool: 5
    replica: true
```

```ruby
# app/models/application_record.rb
class ApplicationRecord < ActiveRecord::Base
  self.abstract_class = true
  connects_to database: { writing: :primary, reading: :primary_replica }
```

```ruby
[1] pry(main)> ActiveRecord::Base.connected_to(role: :reading) { User.first }
ActiveRecord::ReadOnlyError: Write query attempted while in readonly mode: /*application:TestApp,line:(pry):1:in `block in __pry__'*/ SET client_min_messages TO 'warning'
from /app/vendor/bundle/ruby/2.6.0/gems/activerecord-6.0.2.2/lib/active_record/connection_adapters/postgresql/database_statements.rb:85:in `execute'
```

In the above example, the thrown query is `"/*application:TestApp,line:(pry):1:in 'block in __pry__'*/ SET client_min_messages TO 'warning'"`. This is a reading query, but `ActiveRecord::ConnectionAdapters::PostgreSQL::DatabaseStatements#write_query?` fails to return the accurate result. This failure is caused by the presence of comment.

The differences in the results depending on the presence or absence of comments are shown below. When a comment is present at the beginning of SQL, `#write_query?` returns the wrong result. This failure causes the above error.

```ruby
[2] pry(main)> ActiveRecord::Base.connection.write_query?("/*application:TestApp,line:(pry):1:in `block in __pry__'*/ SET client_min_messages TO 'warning'")
=> true  # <- This is wrong

[3] pry(main)> ActiveRecord::Base.connection.write_query?("SET client_min_messages TO 'warning'")
=> false  # <- This is correct
```

`#write_query?` uses a regular expression for checking queries. I think this regular expression doesn't handle comments properly.

https://github.com/rails/rails/blob/bfbf696ed82df38b68331254f65c25813da83559/activerecord/lib/active_record/connection_adapters/postgresql/database_statements.rb#L70-L77

### Expected behavior

`#write_query?` should return false when `"/*<SQL Comment>*/ SET client_min_messages TO 'warning'"` is thrown.

### Actual behavior

`#write_query?` returns true when a comment is inserted at the beginning of SQL.

```ruby
[1] pry(main)> ActiveRecord::Base.connection.write_query?("/*<SQL Comment>*/ SET client_min_messages TO 'warning'")
=> true
```

### System configuration
**Rails version**:
6.0.2.2

**Ruby version**:
2.6.5

**Comment by eugeneius** [comment]
This was addressed by https://github.com/rails/rails/pull/37184, which I've just backported to 6-0-stable. It will be fixed in 6.0.3.

**Comment by south37** [comment]
OK, Thanks! 😄
